homeassistant/components/sensor/ruter.py

```diff
@@ -0,0 +1,92 @@
+"""
+A sensor platform that give you information about next departures from Ruter.
+
+For more details about this platform, please refer to the documentation at
+https://www.home-assistant.io/components/sensor.ruter/
+"""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['pyruter==1.0.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_STOP_ID = 'stop_id'
+CONF_DESTINATION = 'destination'
+CONF_OFFSET = 'offset'
+
+DEFAULT_NAME = 'Ruter'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_STOP_ID): cv.positive_int,
+    vol.Optional(CONF_DESTINATION): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_OFFSET, default=1): cv.positive_int,
+    })
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Create the sensor."""
+    from pyruter.api import Departures
+
+    stop_id = config[CONF_STOP_ID]
+    destination = config.get(CONF_DESTINATION)
+    name = config[CONF_NAME]
+    offset = config[CONF_OFFSET]
+
+    ruter = Departures(hass.loop, stop_id, destination)
```

Comment:
We want all libraries that use aiohttp to take a session argument so that we can reuse the home assistant session throughout the app.

Reply:
Isn't it more like "we would like"? I believe that part of the deal with using third party libraries is that we don't get to dictate their interface.

Still, @ludeeus, please do continue with implementing that :)

Reply:
@amelchio we don't have to dictate their interface, but very much like if they do take in an optional aiohttp session instead of creating their own if they use aiohttp.

Reply:
Yes, and for a library created for Home Assistant that is no biggie. I was more concerned about rejecting a new platform because an existing library does not support passing in the session.

Reply:
Ah no, we wouldn't do that.